### ISSUE TITLE
Some doubts about Context Embedding in model prediction

### ISSUE THREAD

**Issue by lqz72** [Shereen-Elsayed/MBSRec#1]
During the Model Prediction stage, the method mentioned in the paper for model prediction is "we pass the target item id 𝑣𝑜 and the target behavior type 𝑐𝑜 through the same shared layers for item and context embedding, then these embeddings are concatenated and passed through another shared embedding layer to obtain the final target item embedding."

But in my own experiment, I tried to remove the concatenation step and directly used item embedding for prediction, but the performance did not show a significant decrease, so I am very confused about this manifestation. 

Thank you very much for considering my problem. I am looking forward to the possibility of further discussing recommendation system with you.

####
results on Tianchi dataset
- CASM
NDCG@10: 0.5766，HR@10: 0.7426

- CASM w/o target behavior
NDCG@10: 0.5798，HR@10: 0.7419

**Comment by Shereen-Elsayed** [comment]
Thank you for your question; as I understood from your question, you only remove the target context during the test while you keep it during training. 

- It's always better to use the same model settings for both(training and testing) as this can confuse the model. 
- We show in the ablation study that adding the behavior context to the item doesn't significantly impact the model performance, while weighting the items in the loss function affects the performance most.

Hope it helped, thank you again for the discussion.

**Comment by lqz72** [comment]
Thank you for your response. I suggest adding a baseline without weighting the loss function in Tables 4 to 7 of the paper, which will help readers better understand the effectiveness of this method.
